### PR TITLE
stellar-core: 17.0.0 -> 18.5.0

### PR DESCRIPTION
**Pull request by raymens**
###### Description of changes

Update stellar-core to latest version (see https://github.com/stellar/stellar-core/releases/tag/v18.5.0).
Made as a replacement of https://github.com/NixOS/nixpkgs/pull/159781 due to that version being both unable to build with glibc >= 2.34 and it no longer being the latest version.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by raymens** [comment]
Result of `nixpkgs-review pr 167749` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stellar-core</li>
  </ul>
</details>

**Comment by raymens** [comment]
I've ran `stellar-core test` and received the same results as on a ubuntu machine doing the exact same test run on the same release version (18.5.0).

File: pkgs/applications/blockchains/stellar-core/default.nix [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
     owner = "stellar";
     repo = pname;
     rev = "v${version}";
-    sha256 = "1ngl8yjqb8xzhdwzlxzzxf14q2hgwy2ysb17sn5380rrn0jswin1";
+    sha256 = "sha256-wEi22R4zb8d5CJV5eWb776Yob8B6Ok4FrbYI0SGM0H8=";
     fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [ automake autoconf git libtool pkg-config ripgrep ];
+  nativeBuildInputs = [ automake autoconf git libtool pkg-config ripgrep libunwind ];
```
Comment (Artturin) [inline pkgs/applications/blockchains/stellar-core/default.nix:16]: this should preferably be

```suggestion
  nativeBuildInputs = [ autoreconfHook git pkg-config ripgrep libunwind ];
```

https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#autoconf-setup-hook-autoconf

Comment (Artturin) [reply pkgs/applications/blockchains/stellar-core/default.nix:16]:
```suggestion
  nativeBuildInputs = [ automake autoconf git libtool pkg-config ripgrep ];
```

`libunwind` is a library so it should be in `buildInputs`

Comment (raymens) [reply pkgs/applications/blockchains/stellar-core/default.nix]: Switching to autoreconfHook is causing build failures I'm currently investigating. I've updated the file with the libunwind changes though.

**Comment by raymens** [comment]
Result of `nixpkgs-review pr 167749` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stellar-core</li>
  </ul>
</details>

**Pull request merged**